src/Syncs/BloomFilter.cpp

```diff
@@ -0,0 +1,160 @@
+/* This code is part of the GenSync project developed at Boston University.  Please see the README for use and references. */
+
+//
+// Created by Anish Sinha on 12/4/23.
+//
+
+#include <GenSync/Syncs/BloomFilter.h>
+
+BloomFilter::BloomFilter() = default;
+BloomFilter::~BloomFilter() = default;
+
+BloomFilter::BloomFilter(size_t size, int nHash)
+{
+    this->bfSize = size;
+    this->bits.resize(size, 0);
+    this->numHashes = nHash;
+}
+
+BloomFilter::BloomFilter(size_t numExpElems, float falsePosProb)
+{
+    if(falsePosProb < 0 || falsePosProb > 1)
+    {
+        falsePosProb = 0.05;
+    }
+    
+    this->bfSize = round(-2.08 * log(falsePosProb) * numExpElems);
+    this->bits.resize(this->bfSize, 0);
+    this->numHashes = round(-2.08 * log(falsePosProb) * log(2));
+}
+
+size_t BloomFilter::getSize()
+{
+    return this->bfSize;
+}
+
+int BloomFilter::getNumHashes()
+{
+    return this->numHashes;
+}
+
+vector<bool> BloomFilter::getBits()
+{
+    return this->bits;
+}
+
+float BloomFilter::getFalsePosProb(size_t numExpElems)
+{
+    float exp = ((float)bfSize/numExpElems) * log(2);

```

Comment:
You're using this formula a lot ... perhaps you should have a (commented) private method that does this?

Reply:
I have added the comment to cite the paper which I read the formula from. I am unsure how to make this formula into a private method because for each time it is used, the calculation is slightly different. For example, I use it to calculate the optimal number of hashes in one instance and then the false positive probability in another.